waiter/integration/waiter/token_request_test.clj

```diff
@@ -1366,6 +1366,76 @@
         (finally
           (delete-token-and-assert waiter-url token))))))
 
+(deftest ^:parallel ^:integration-fast test-maintenance-mode
+  (testing-using-waiter-url
+    (let [service-name (rand-name)
+          token (create-token-name waiter-url service-name)
+          current-user (retrieve-username)
+          service-description (-> (kitchen-request-headers :prefix "")
+                                  (assoc :name service-name
+                                         :permitted-user "*"
+                                         :run-as-user current-user))
+          token-root (retrieve-token-root waiter-url)
+          token-cluster (retrieve-token-cluster waiter-url)
+          custom-maintenance-message "custom maintenance message"
+          token-maintenance {:message custom-maintenance-message}
+          request-headers {:x-waiter-token token}]
+      (try
+        (testing "token created in maintenance mode"
+          (let [token-description (assoc service-description
+                                    :maintenance token-maintenance
+                                    :token token)
+                response (post-token waiter-url token-description)]
+            (assert-response-status response http-200-ok)))
+
+        (testing "token retrieval"
+          (let [token-response (get-token waiter-url token)
+                response-body (-> token-response :body json/read-str pc/keywordize-map)]
+            (is (contains? response-body :last-update-time))
+            (is (= (assoc service-description
+                     :cluster token-cluster
+                     :last-update-user current-user
+                     :maintenance token-maintenance
+                     :owner current-user
+                     :previous {}
+                     :root token-root)
+                   (dissoc response-body :last-update-time)))))
+
+        (testing "index should be updated with maintenance enabled"
+          (let [response (make-request waiter-url "/tokens")
+                body (-> response :body json/read-str walk/keywordize-keys)
+                index (first body)]
+            (is (true? (:maintenance index)))))
+
+        (testing "request to service should error with custom maintenance message"
+          (let [{:keys [body] :as response}
+                (make-request waiter-url "/hello-world" :headers request-headers)]
+            (assert-response-status response http-503-service-unavailable)
```

Comment:
Can you include, in the PR description, an example of what the request log looks like for this request?